Branch-SDK/src/io/branch/referral/RemoteInterface.java

```diff
@@ -173,13 +179,22 @@ private ServerResponse make_restful_get(String baseUrl, JSONObject params, Strin
         }
 
         try {
+            modifiedUrl = getResolvedApiUrl(modifiedUrl);
             if (log) PrefHelper.Debug("BranchSDK", "getting " + modifiedUrl);
             lastRoundTripTime_ = 0;
             long reqStartTime = System.currentTimeMillis();
             URL urlObject = new URL(modifiedUrl);
-            connection = (HttpURLConnection) urlObject.openConnection();
+            connection = (HttpsURLConnection) urlObject.openConnection();
             connection.setConnectTimeout(timeout);
             connection.setReadTimeout(timeout);
+            // Default host verifier will provide SSL error on host verification since ip used in request is not mapped to a certificate
+            connection.setHostnameVerifier(new HostnameVerifier() {
```

Comment:
Android internally check certificate for any secure connection. The way it find the SSL certificate is using the CommonName of the host. When we change the host with its IP Android won't be able to  verify SSL and throw `hostNotVerified` exception. This code will override Android SSL check and return it as always verified